src/drivers/dw/dma.c

```diff
@@ -831,17 +841,10 @@ static void dw_dma_verify_transfer(struct dma_chan_data *channel,
 
 		dcache_writeback_region(dw_chan->lli,
 					sizeof(struct dw_lli) * channel->desc_count);
-#else
-		while (lli->ctrl_hi & DW_CTLH_DONE(1)) {
-			lli->ctrl_hi &= ~DW_CTLH_DONE(1);
-			dw_chan->lli_current =
-				(struct dw_lli *)dw_chan->lli_current->llp;
-			lli = platform_dw_dma_lli_get(dw_chan->lli_current);
-		}
-#endif
```

Comment:
Everything else looks good, not sure why this is going away though. What was the logic here and why is this no longer needed?

Reply:
@keyonjie this code clears all the DONE bits in the LLI that's be written by HW. If these bits are not clear then DMA will stop.

Reply:
@paulstelian97 @lgirdwood this part is covered by above Zephyr lines:
```
		for (i = 0; i &lt; channel-&gt;desc_count; i++)	
			dw_chan->lli[i].ctrl_hi &= ~DW_CTLH_DONE(1);
```

Reply:
> Sorry I mean, when we dont use Zephyr ?

before change,
Zephyr case: do invalidate -> write -> writeback
Non-Zephyr case: just write.
after my change,
unified for Zephyr and Non-Zephyr cases('#if defined __ZEPHYR__' is removed): do invalidate -> write -> writeback

Reply:
This is racy and there is no easy way around it. We are racing against the DMAC writing back the LLI data which could occur during this loop. This may be safer if we allocate LLI as an uncache region. i.e. both DW driver and DMAC use the uncache address for LLI.